addons/account/wizard/account_tour_upload_bill.py

```diff
@@ -23,18 +30,80 @@ class AccountTourUploadBill(models.TransientModel):
     )
 
     def _selection_values(self):
+        values = [
+            ('sample', _('Try a sample vendor bill')),
+            ('upload', _('Upload your own bill')),
+        ]
         journal_alias = self.env['account.journal'] \
             .search([('type', '=', 'purchase'), ('company_id', '=', self.env.company.id)], limit=1)
+        if journal_alias.alias_domain:
+            values += [('email', _('Or send a bill to %s@%s', journal_alias.alias_name, journal_alias.alias_domain))]
+        else:
+            values += [('noemail', _('Or send a bill by email'))]
 
-        return [('sample', 'Try a sample vendor bill'),
-                ('upload', 'Upload your own bill'),
-                ('email', 'Or send a bill to %s@%s' % (journal_alias.alias_name, journal_alias.alias_domain))]
+        return values
 
     def _compute_sample_bill_image(self):
         """ Retrieve sample bill with facturx to speed up onboarding """
+        def addAddress(canvas, x, y, partner):
+            can.setFillColorRGB(1, 1, 1)
+            can.rect(x, y, width*0.3, height*0.1, fill=1, stroke=0)
```

Comment:
the `can` variable is really weird. It look like a programmation error but it works like a miracle because the `can` variable is defined out of this method scope. Use the `canvas` variable instead!

Reply:
renamed the functiion argument